create.ps1

```diff
@@ -1,189 +1,22 @@
-###########################################
+#################################################
 # HelloID-Conn-Prov-Target-Inception-Create
-#
-# Version: 1.0.2
-###########################################
-# Initialize default values
-$config = $configuration | ConvertFrom-Json
-$p = $person | ConvertFrom-Json
-$success = $false
-$auditLogs = [System.Collections.Generic.List[PSCustomObject]]::new()
-
-# Script Configuration
-$departmentLookupProperty = { $_.Department.ExternalId }
-$titleLookupProperty = { $_.Title.ExternalId }
-
-#Generate surname conform nameconvention
-function Get-LastName {
-    [CmdletBinding()]
-    param (
-        [Parameter(Mandatory)]
-        [ValidateNotNullOrEmpty()]
-        [object]
-        $person
-    )
-
-    if ([string]::IsNullOrEmpty($person.Name.FamilyNamePrefix)) {
-        $prefix = ""
-    }
-    else {
-        $prefix = $person.Name.FamilyNamePrefix + " "
-    }
-
-    if ([string]::IsNullOrEmpty($person.Name.FamilyNamePartnerPrefix)) {
-        $partnerPrefix = ""
-    }
-    else {
-        $partnerPrefix = $person.Name.FamilyNamePartnerPrefix + " "
-    }
-
-    $Surname = switch ($person.Name.Convention) {
-        "B" { $person.Name.FamilyName }
-        "BP" { $person.Name.FamilyName + " - " + $partnerprefix + $person.Name.FamilyNamePartner }
-        "P" { $person.Name.FamilyNamePartner }
-        "PB" { $person.Name.FamilyNamePartner + " - " + $prefix + $person.Name.FamilyName }
-        default { $prefix + $person.Name.FamilyName }
-    }
-
-    $Prefix = switch ($p.Name.Convention) {
-        "B" { $prefix }
-        "BP" { $prefix }
-        "P" { $partnerPrefix }
-        "PB" { $partnerPrefix }
-        default { $prefix }
-    }
-    $output = [PSCustomObject]@{
-        surname  = $Surname
-        prefixes = $Prefix.Trim()
-    }
-    Write-Output $output
-}
-
-# Employee Account mapping
-$account = [PSCustomObject]@{
-    staffnumber          = $p.ExternalId
-    firstname            = $p.Name.NickName
-    lastname             = (Get-LastName -Person $p).surname
-    middlename           = (Get-LastName -Person $p).prefixes
-    initials             = $p.Name.Initials
-    email                = $p.Contact.Business.Email
-    phone                = $p.Contact.Business.Phone.Fixed
-    dateofbirth          = if ($null -ne $p.Details.BirthDate ) { '{0:yyyy-MM-dd}' -f ([datetime]$p.Details.BirthDate) };
-    startdate            = if ($null -ne $p.PrimaryContract.StartDate ) { '{0:yyyy-MM-dd}' -f ([datetime]$p.PrimaryContract.StartDate) };
-    enddate              = $null # ((Get-Date).AddDays(-1))
-    positionsPerOrgUnits = @()
-}
-
+# PowerShell V2
+# Version: 1.0.0
```

Comment:
Version needs to be 2.0.0. But can also be removed (like in template)

The PSv1 will be "release" v1.0.2. This code should be "release" v2.0.0

Reply:
updated version